ols/app/models/config.py

```diff
@@ -109,6 +109,56 @@ def validate_yaml(self) -> None:
             )
 
 
+class TLSConfig(BaseModel):
+    """TLS configuration."""
+
+    ssl_certfile: Optional[str] = None
+    ssl_keyfile: Optional[str] = None
+    ssl_keyfile_password: Optional[str] = None
+
+    def __init__(self, data: Optional[dict] = None) -> None:
+        """Initialize configuration and perform basic validation."""
+        super().__init__()
+        if data is not None:
+            self.ssl_certfile = data.get("ssl_certfile", self.ssl_certfile)
+            self.ssl_keyfile = data.get("ssl_keyfile", self.ssl_keyfile)
+            self.ssl_keyfile_password = data.get(
+                "ssl_keyfile_password", self.ssl_keyfile_password
+            )
+
+    def validate_yaml(self) -> None:
```

Comment:
There're 8 combinations to consider. It doesn't make sense to not have a cert file, so we need to check the remaining 4.

```
def _file_check(path: str, desc: str) -> None:
    """Check that path is a readable regular file."""
    if not os.path.exists(path):
        raise InvalidConfigurationError(f"{desc} '{path}' does not exist")
    if not os.path.isfile(path):
        raise InvalidConfigurationError(f"{desc} '{path}' is not a directory")
    if not os.access(path, os.R_OK):
        raise InvalidConfigurationError(f"{desc} '{path}' is not readable")


def validate_yaml(self) -> None:
    """Validate TLS config."""
    if not self.ssl_certfile:
        raise InvalidConfigurationError(
            "Can not enable TLS without ols_config.tls_config.ssl_certfile"
        )
    else:
        _file_check(self.ssl_certfile, "OLS server certificate")
        if self.ssl_keyfile:
            _file_check(self.ssl_keyfile, "OLS server certificate private key")
            if self.ssl_keyfile_password:
                _file_check(self.ssl_keyfile_password, "OLS server certificate private key password")
        else:
            if self.ssl_keyfile_password:
                raise InvalidConfigurationError(
                    "Can not have ols_config.tls_config.ssl_certfile_password without ols_config.tls_config.ssl_keyfile"
                )
```

Reply:
we need only to check that if we provide a cert/key that they exist because they get used as a pth in the runtime

for passwod, i have modified it to be read from a file, instead of directly seting the pass in the config.yaml

i followed same logic as we do with the OPENAI_KEY config

Reply:
There's no question about that. My comment was more about checking for valid combinations of ssl_certfile, ssl_keyfile and ssl_keyfile_password.

Reply:
the proper check for the cert/key value is that if one value is provided, the other value must also be provided.

we have similar validation checks for other coupled fields.

(either both are None, or both have a valid filepath specified, or it's an error)

and yes it's also an error if they don't provide TLS key/cert values, but TLS is not disabled.  But i see that check is already being done during startup, which is fine.